src/Acuminator/Acuminator.Utilities/DiagnosticSuppression/SuppressionManager.cs

```diff
@@ -88,53 +83,82 @@ public static void Init(ISuppressionFileSystemService fileSystemService, IEnumer
 
 		public static void SaveSuppressionBase()
 		{
-			if (Instance == null)
+			CheckIfInstanceIsInitialized();
+			
+			lock (Instance._fileSystemService)
 			{
-				throw new InvalidOperationException($"{nameof(SuppressionManager)} instance was not initialized");
-			}
+				//Create local copy in order to avoid concurency problem when the collection is changed during the iteration
+				var filesWithGeneratedSuppression = Instance._fileByAssembly.Values.Where(f => f.GenerateSuppressionBase).ToList();
 
-			var filesWithGeneratedSuppression = Instance._fileByAssembly.Values.Where(f => f.GenerateSuppressionBase);
-
-			foreach (var file in filesWithGeneratedSuppression)
-			{
-				Instance._fileSystemService.Save(file.MessagesToDocument(), file.Path);
+				foreach (var file in filesWithGeneratedSuppression)
+				{
+					Instance._fileSystemService.Save(file.MessagesToDocument(), file.Path);
+				}
 			}
 		}
 
-		private bool IsSuppressed(SemanticModel semanticModel, Diagnostic diagnostic, CancellationToken cancellation)
+		public static SuppressionFile CreateSuppressionFileForProject(Project project)
 		{
-			cancellation.ThrowIfCancellationRequested();
+			project.ThrowOnNull(nameof(project));
+			CheckIfInstanceIsInitialized();
 
-			var (assembly, message) = GetSuppressionInfo(semanticModel, diagnostic, cancellation);
+			//First check if file already exists to dismiss threads withou acquiring the lock
+			if (Instance._fileByAssembly.TryGetValue(project.Name, out var existingSuppressionFile) && existingSuppressionFile != null)
+				return existingSuppressionFile;
 
-			if (assembly == null)
+			lock (Instance._fileSystemService)
```

Comment:
@SENya1990 why do we need a look here?

Reply:
@metallmonster because the VS throws exception about access to file from another thread.